rita/src/rita_client/exit_manager/mod.rs

```diff
@@ -269,11 +272,17 @@ impl Handler<Tick> for ExitManager {
             trace!("We have selected an exit!");
             if let Some(ref general_details) = exit.info.general_details() {
                 trace!("We have details for the selected exit!");
-                if let Some(_) = exit.info.our_details() {
+                // only run if we have our own details and we either have no setup exit or the chosen
+                // exit has changed
+                if exit.info.our_details().is_some()
+                    && !(self.last_exit.is_some() && self.last_exit.clone().unwrap() == exit)
+                {
                     trace!("We are signed up for the selected exit!");
                     linux_setup_exit_tunnel().expect("failure setting up exit tunnel");
                     TrafficWatcher::from_registry()
                         .do_send(Watch(exit.id.clone(), general_details.exit_price));
```

Comment:
This should run every time, as traffic needs to be settled every 5 seconds, not only when exits change